x/ibc/handler.go

```diff
@@ -195,16 +195,7 @@ func NewHandler(k Keeper) sdk.Handler {
 			if !ok {
 				return nil, sdkerrors.Wrapf(port.ErrInvalidRoute, "route not found to module: %s", module)
 			}
-			res, err := cbs.OnTimeoutPacket(ctx, msg.Packet)
-			if err != nil {
-
-				return nil, err
-			}
-			err = k.ChannelKeeper.TimeoutExecuted(ctx, cap, msg.Packet)
```

Comment:
is it secure to move deletion/capability check to ante handler calling of AcknowledgePacket/TimeoutPacket?

Reply:
As long as the message processing is atomic either way, I think it's fine.